### ISSUE TITLE
Reverting the graphs

### ISSUE THREAD

**Issue by aravindadithya** [aravindadithya/DLR#30]
Imagine training runs for 20 epochs, with the best validation accuracy at epoch 15. If training continues for another 20 epochs, the code will reload the best weights from epoch 15 and redo epochs 16 to 20, but these get logged as epochs 21 to 25. Ideally, all graphs should roll back to epoch 15, and data from epochs 16 to 20 should be removed. This can be handled with the wandb revert feature, though the tricky part is syncing graphs that use epochs on the x-axis like accuracy, loss etc.. with those that use steps, like parameters or gradients.

**Comment by aravindadithya** [comment]
Implement more logs for
1. Log Saliency maps( where the network tries to look)
2. Ablation studies
3. Effect of covariance matrix of weight filters on the images. 
4. Log initialization technique
